LevelComponents/Entity.cpp

```diff
@@ -255,6 +255,100 @@ namespace LevelComponents
+    /// </summary>
+    /// <param name="oamNum">
+    /// the number of oam.
+    /// </param>
+    /// <param name="nakedOAMarray">
+    /// Naked OAM array without oam number as the first element
+    /// </param>
+    /// <param name="noReferrneceBox">
+    /// Don't draw the yellow referrence Box if set this to true
+    /// </param>
+    QImage Entity::TestRenderOams(int oamNum, unsigned short *nakedOAMarray, bool noReferrneceBox)
+    {
+        if (!oamNum) return QImage();
+
+        QVector<OAMTile *> tmpOAMTiles;
+        for (int i = 0; i < oamNum; ++i)
+        {
+            // Obtain short values for the OAM tile
+            unsigned short attr0 = nakedOAMarray[3 * i];
+            unsigned short attr1 = nakedOAMarray[3 * i + 1];
+            unsigned short attr2 = nakedOAMarray[3 * i + 2];
+            // skip priority reset and Char Id (tiles ID) reset
+            // skip orientation change
+            // skip affine transformation (double size)
+            // skip alpha blend
+
+            // Obtain the tile parameters for the OAM tile
+            struct OAMTile *newOAM = new struct OAMTile();
+            newOAM->Xoff = (attr1 & 0xFF) - (attr1 & 0x100); // Offset of OAM tile from entity origin
+            newOAM->Yoff = (attr0 & 0x7F) - (attr0 & 0x80); // they are signed char
+            newOAM->xFlip = (attr1 & (1 << 0xC)) != 0;
+            newOAM->yFlip = (attr1 & (1 << 0xD)) != 0;
+            int SZ = (attr1 >> 0xE) & 3;                         // object size
+            int SH = (attr0 >> 0xE) & 3;                         // object shape
+            newOAM->OAMwidth = OAMDimensions[2 * (SZ * 3 + SH)]; // unit: 8x8 tiles
+            newOAM->OAMheight = OAMDimensions[2 * (SZ * 3 + SH) + 1];
+            int tileID = attr2 & 0x3FF;
+            int palNum = (attr2 >> 0xC) & 0xF;
+
+            // Create the tile8x8 objects
+            for (int y = 0; y < newOAM->OAMheight; ++y)
+            {
+                for (int x = 0; x < newOAM->OAMwidth; ++x)
+                {
+                    struct EntityTile *entityTile = new struct EntityTile();
+                    entityTile->deltaX = x * 8;
+                    entityTile->deltaY = y * 8;
+                    int offsetID = tileID + y * 0x20 + x;
+                    if (offsetID >= 0x200) offsetID = offsetID - 0x200; // the game engine use char Id with offset in the oam data
+                    if (offsetID >= tile8x8data.size()) continue; // this data have problems, just skip it
+                    if (tile8x8data[offsetID] == blankTile) continue; // skip dummy tiles if used
+                    Tile8x8 *newTile = new Tile8x8(tile8x8data[offsetID]);
+                    if (palNum > 7) palNum = palNum - 8; // the game engine use pal Id with offset in the oam data
+                    newTile->SetPaletteIndex(palNum);
+                    entityTile->objTile = newTile;
+                    newOAM->tile8x8.push_back(entityTile);
```

Comment:
Where does entityTile get used later? This also seems like a memory leak and does not need to be a pointer

Reply:
it is deleted in the deconstructor of OAMTile which can be deleted when deleting OAMTile, which is done in the render for loop